### DISCUSSION TITLE
[mini.bracketed] bracketed mapping to cycle through errors in quickfix?

### DISCUSSION THREAD

**Discussion by beaumccartney** [nvim-mini/mini.nvim#1185] [Q&A]
presently, ]q and [q cycle through every line of the quickfix, including those that don't match errorformat.

I end up having to mash to get to errors this way. Is there existing functionality to cycle through just the errors?

**Comment by echasnovski** [comment]
The `quickfix` target in 'mini.bracketed' is essentially a [thin wrapper around `vim.fn.getqflist()` function and `:cc` command](https://github.com/echasnovski/mini.nvim/blob/57e47cf7a2923684e7413989ab267ed9730e7d03/lua/mini/bracketed.lua#L1782-L1815). It *is* possible to adjust it to iterate only through errors by adjusting `next` and `prev` fields of an iterator ([`MiniBracketed.advance()`](https://github.com/echasnovski/mini.nvim/blob/57e47cf7a2923684e7413989ab267ed9730e7d03/doc/mini-bracketed.txt#L607) is exported).

What is lacking (in my opinion) is a definitive rule that can tell if quickfix entry is for error or not. There is a `type` field which is documented as "type of the error, 'E', '1', etc.", but I am not sure how reliable this is. While I also not really sure how to work with 'errorformat'.

All in all, the answer is "Yes, this should be doable. But I am afraid it won't be built-in and requires more research for me to actually implement than I'd like to do".
